### ISSUE TITLE
[HOLD for payment 2024-12-20] [$250] Add Chat with setup specialist to the accounting page.

### ISSUE THREAD

**Issue by danielrvidal** [Expensify/App#51487]
Problem: Self-onboarding users who navigate to their accounting integrations and don’t see their accounting package have no idea we can support their use case.

Solution:  Add a note beneath the accounting options to talk to their onboarding specialist to help users get their accounting setup if they don't see their account package. 

The page below is getting updated with the following options and logic: 
- If they have an onboarding specialist assigned display the following and link to the #admins room with the setup specialist.
>[?] Need another accounting package?                          [Talk to your onboarding specialist.]
- If not, if they have an account manager assigned display the following and link to the DM with their account manager.
> [?] Need another accounting package?                          [Talk to your account manager.]
- Else, display the following and link to their Concierge DM.
> [?] Need another accounting package?                          [Talk to concierge.]

# Existing 
![image](https://github.com/user-attachments/assets/d8677772-0a75-44ff-896d-b4835e5fca99)

# Proposed
Here is what the updated version looks like assuming they have an onboarding specialist.
![image](https://github.com/user-attachments/assets/6e8312ee-fd87-4d48-b2dd-970b5c41a2ad)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021851395014687053617</li>
        <li>Upwork Job ID: 1851395014687053617</li>
        <li>Last Price Increase: 2024-11-12</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Reviewer | 104980673</li>
<li>nkdengineer | Contributor | 104980674</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @danielrvidal</details>

**Comment by danielrvidal** [comment]
I think this one is good to go. I'll update this and get someone from Design to help out with the mocks.

**Comment by danielrvidal** [comment]
I put the latest update in here: https://expensify.slack.com/archives/C07HPDRELLD/p1730158078039199?thread_ts=1729874539.081569&cid=C07HPDRELLD

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Add Chat with setup specialist to the accounting page.

### What is the root cause of that problem?

This is a new feature request
### What changes do you think we should make in order to solve the problem?
> If they have an onboarding specialist assigned display the following and link to the #admins room with the setup specialist.
[?] Need another accounting package? [Talk to your onboarding specialist]

> If not, if they have an account manager assigned display the following and link to the DM with their account manager.
[?] Need another accounting package? [Talk to your account manager]

> Else, display the following and link to their Concierge DM.
[?] Need another accounting package? [Talk to concierge]


We can add a note beneath after this link like this. We need to create a new translation for each text note and add the text and the link inside the text link accordingly for each case above. This can be done in the PR phrase

```
<View style={[styles.flexRow, styles.alignItemsCenter, styles.mt2]}>
    <Icon
        src={Expensicons.QuestionMark}
        width={20}
        height={20}
        fill={theme.icon}
        additionalStyles={styles.mr3}
    />
    <View style={[isSmallScreenWidth || isMediumScreenWidth ? styles.flexColumn : styles.flexRow]}>
        <Text>Need another accounting package? </Text>
        // The text inside text link will be changed for each case
        <TextLink href={}>Talk to your onboarding specialist</TextLink>
    </View>
</View>
```

https://github.com/Expensify/App/blob/dad3f63c31b3d585b892e93ea6620751e453311f/src/pages/workspace/accounting/PolicyAccountingPage.tsx#L545

### What alternative solutions did you explore? (Optional)

**Comment by nkdengineer** [comment]
> onboarding specialist assigned 

> an account manager assigned

@danielrvidal How can we get this data in Onyx?

**Comment by danielrvidal** [comment]
@trjExpensify do you know if it's possible for them to get the information in Onyx, or will I need to work with an internal engineer for that?

**Comment by trjExpensify** [comment]
Yeah, I don't think it's available in Onyx right now. We might be able to get it by looking at the participants of the #admins room -- but I don't see where we distinguish between Guide or AM, nor can we fully rely on the chat data being loaded with focus mode (CC: @mountiny).  

I was looking at the code for the "Support methods" modal in OldDot as well to see how that's done [here](https://github.com/Expensify/Web-Expensify/blob/035404c9cd8167d37452a6471d63fbdfee6528b0/site/app/modals/userMenu/helpAndFeedbackView.jsx#L62-L91)

<img width="995" alt="image" src="https://github.com/user-attachments/assets/c210deff-454b-45a6-9f6d-70941ecb33fd">

Seems like there's `getAssignedSupportData` in Web-E to extrapolate if there's an assigned guide, account manager, or partner manager: https://github.com/Expensify/Web-Expensify/blob/035404c9cd8167d37452a6471d63fbdfee6528b0/site/lib/store/userStore.jsx#L149-L160

We might be able to use something like that, but I think we'll need someone internal to help look at this.

**Comment by danielrvidal** [comment]
@mountiny do you mind commenting next week when you get a chance? Thank you!

**Comment by mountiny** [comment]
We could load the nvps which include who is the guide or AM for the user and link accordingly. I am not sure though if the guide/ AM is assigned already before they are technically needed, so it also might not be the only datapoint we need.

@techievivek I believe you have great understanding of how the guides/ ams assignment works. What do you think, what would be the best way to achieve the OG requirement

**Comment by techievivek** [comment]
Yeah, as Tom mentioned, we can either reuse the `getAccountManagerAndGuideData` or `getAssignedSupportData` API command to fetch the support data. This will require updating the code to push the Onyx updates to the requesting clients.

**Comment by danielrvidal** [comment]
So @techievivek I'm assuming that means we'll need to have an internal engineer take that part, right? I do think it would be a good update for our contributors to be able to access.

**Comment by techievivek** [comment]
Yeah, I can work on the internal changes for this.

**Comment by techievivek** [comment]
Working on adding a new command to make it 1:1:1 and also to queue the onyx updates from the Auth.

**Comment by techievivek** [comment]
@dukenv0307 @mountiny Should we add a new Onyx key or just add the support-related data in the account or some other key? I don't have a strong preference here so just seeing if you guys have any thoughts, otherwise I can add it as a new key.

**Comment by mountiny** [comment]
if its per account, makes sense to put it in to the account key

**Comment by techievivek** [comment]
Auth PR is in review. https://github.com/Expensify/Auth/pull/13054

**Comment by techievivek** [comment]
Not overdue, Backend PR is up for review.

**Comment by techievivek** [comment]
Auth PR is merged yesterday, working on the web-e PR now.

**Comment by danielrvidal** [comment]
Nice, looks like regular updates in the issues to update the Backend. Let's keep pressing so we can get this one done this week!

**Comment by techievivek** [comment]
Bumped for the review.

**Comment by techievivek** [comment]
We can likely work on the frontend changes in parallel. If needed, I can share a link to my NGROK instance for testing. However, sharing the API response alone should also be helpful in this case.

**Comment by techievivek** [comment]
This is what the Onyx response via HTTP would look like. We are not pushing any onyx update for this rather just returning the update via HTTP call.

method : merge
key: account
value:
```
 {"accountManagerReportID": <DM chat reportID between user and their account manager>},
 {"accountManagerAccountID": <accountID>},
{"accountManagerEmail": <email>},
{"reportIDWithPartnerManager": <DM chat reportID between user and their partner manager>},
{"partnerManagerAccountID": <accountID>},
{"partnerManagerEmail": <email>},
{"adminsRoomReportID": <admins room reportID corresponding to active policy>},
{"guideDetails": {"guideEmail": <guide email>, "wonTimestamp": <datetime stamp if the user is already onboard> }}
```

**Comment by techievivek** [comment]
@nkdengineer Do you have an updated proposal based on above info?

**Comment by nkdengineer** [comment]
@techievivek  Updated [proposal](https://github.com/Expensify/App/issues/51487#issuecomment-2446168063).

**Comment by nkdengineer** [comment]
@dukenv0307 The proposal is ready to review again.

**Comment by techievivek** [comment]
Thanks for the updated proposal, the command that we will be calling here is `GetAssignedSupportData`.

**Comment by jamesdeanexpensify** [comment]
@danielrvidal just caught this one, looking at the OP up top: 

- We should say `setup specialist` instead of `onboarding specialist` since we've standardized on that, and also `Concierge` instead of `concierge` 
- I also wonder if we should say 'Chat with your...' instead of `Talk with your...` to promote chat

**Comment by danielrvidal** [comment]
@nkdengineer do you mind updating the proposal to say `Chat with` rather than `Talk to`? cc @dukenv0307

**Comment by jamesdeanexpensify** [comment]
Including copy updates in the same thread as Spanish translations [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1732219484481979?thread_ts=1732164691.179079&cid=C01GTK53T8Q), FYI!

**Comment by nkdengineer** [comment]
I'll update soon

**Comment by nkdengineer** [comment]
@danielrvidal @jamesdeanexpensify @dukenv0307 i updated translations in open PR [here](https://github.com/Expensify/App/pull/52803)

**Comment by jamesdeanexpensify** [comment]
Thank you!

**Comment by techievivek** [comment]
Regression test looks good to me. @danielrvidal can we please get this added to testRail, thanks.

**Comment by danielrvidal** [comment]
I'm guessing this will happen when break is done. I don't think we need it done immediately though.

**Comment by techievivek** [comment]
Not overdue, payment will be done when team has returned from the break.

**Comment by danielrvidal** [comment]
Sounds good, we'll pick it back up next week.

**Comment by stephanieelliott** [comment]
Summarizing payment on this issue: 

 - Contributor: @nkdengineer $250 via Upwork -- PAID 
 - Contributor+: @dukenv0307 $250 via ND - please request!
 
 Upwork job is here: https://www.upwork.com/jobs/~021851395014687053617

**Comment by JmillsExpensify** [comment]
$250 approved for @dukenv0307
